aspnetcore/security/cors.md

```diff
@@ -3,181 +3,231 @@ title: Enable Cross-Origin Requests (CORS) in ASP.NET Core
 
-[!code-csharp[](cors/sample/CorsExample4/Startup.cs?range=53-58)]
+To allow all author request headers, call <xref:Microsoft.AspNetCore.Cors.Infrastructure.CorsPolicyBuilder.AllowAnyHeader*>:
 
-To allow all author request headers:
+[!code-csharp[](cors/sample/CorsExample4/Startup.cs?range=63-68&highlight=5)]
 
-[!code-csharp[](cors/sample/CorsExample4/Startup.cs?range=62-67)]
+This setting affects [preflight requests and the Access-Control-Request-Headers header](#preflight-requests) (described later in this topic).
 
-Browsers are not entirely consistent in how they set Access-Control-Request-Headers. If you set headers to anything other than "*", you should include at least "accept", "content-type", and "origin", plus any custom headers that you want to support.
+::: moniker range=">= aspnetcore-2.2"
 
-### Set the exposed response headers
+A CORS Middleware policy match to specific headers specified by `WithHeaders` is only possible when the headers sent in `Access-Control-Request-Headers` exactly match the headers stated in `WithHeaders`.
+
+For instance, consider an app configured as follows:
 
-By default, the browser doesn't expose all of the response headers to the application. (See [http://www.w3.org/TR/cors/#simple-response-header](http://www.w3.org/TR/cors/#simple-response-header).) The response headers that are available by default are:
+```csharp
+app.UseCors(policy => policy.WithHeaders(HeaderNames.CacheControl));
+```
+
+CORS Middleware declines a preflight request with the following request header because `Content-Language` ([HeaderNames.ContentLanguage](xref:Microsoft.Net.Http.Headers.HeaderNames.ContentLanguage)) isn't listed in `WithHeaders`:
+
+```
+Access-Control-Request-Headers: Cache-Control, Content-Language
+```
+
+The app returns a *200 OK* response but doesn't send the CORS headers back. Therefore, the browser doesn't attempt the cross-origin request.
+
+::: moniker-end
+
+::: moniker range="< aspnetcore-2.2"
+
+CORS Middleware always allows four headers in the `Access-Control-Request-Headers` to be sent regardless of the values configured in CorsPolicy.Headers. This list of headers includes:
+
+* `Accept`
+* `Accept-Language`
+* `Content-Language`
+* `Origin`
+
+For instance, consider an app configured as follows:
+
+```csharp
+app.UseCors(policy => policy.WithHeaders(HeaderNames.CacheControl));
+```
 
-* Cache-Control
+CORS Middleware responds successfully to a preflight request with the following request header because `Content-Language` is always whitelisted:
 
-* Content-Language
+```
+Access-Control-Request-Headers: Cache-Control, Content-Language
+```
 
-* Content-Type
+::: moniker-end
 
-* Expires
+### Set the exposed response headers
+
+By default, the browser doesn't expose all of the response headers to the app. For more information, see [W3C Cross-Origin Resource Sharing (Terminology): Simple Response Header](http://www.w3.org/TR/cors/#simple-response-header).
```

Comment:
@scottaddie Is it ok if I make *all* of the links in the topic `https`? Currently, the examples (e.g., `http://myservice.azurewebsites.net/api/test`, `http://www.example.com/home`) all go `http`, and I think it would ok to make them all secure.

Reply:
Except in the *Same origin* section. It has to show a different protocol there to explain the same origin concept. Elsewhere tho ... I'd like to make all of the links secure.

Reply:
I agree it makes sense everywhere except for the Same origin section.